### ISSUE TITLE
feature request: filter output based on error code

### ISSUE THREAD

**Issue by keunes** [linkcheck/linkchecker#70]
Maybe this is already possible, but I couldn't find it on the [man page](https://linkcheck.github.io/linkchecker/man1/linkchecker.1.html).
I've been handed a report by the tech team in my office, which include quite a few 403 errors. When processing the link report, it would save a great deal of time if those errors weren't included. :)
It's already possible to ignore specific URLs from for the check; for me it would be equally helpful if the tech team could have excluded 403 errors from the report.

**Comment by lembergerth** [comment]
We have a similar issue: We have a large amount of internal, password-protected webpages that all have different passwords and don't stick to a single URL-pattern. Instead of going through the hassle to specify the credentials for each of these, I would prefer to simply ignore all *401 Unauthorized* errors.
Linkchecker has the ability to ignore warnings, but it would be nice to be able to also ignore certain errors.

**Comment by cjmayo** [comment]
I imagine some people would consider these errors, so we can't convert them to be warnings, probably something like c) but perhaps `ignoreerrors`.

For your use case though isn't the existing `ignore` enough? If these URLs are always failing with linkchecker?

Ultimately though, although it is fine to raise feature requests, new features are only likely to come about from PRs!

**Comment by lpirl** [comment]
> I imagine some people would consider these errors, so we can't convert them to be warnings, probably something like c) but perhaps `ignoreerrors`.

Yes, this would also work. For a moment, I thought "What if we want to ignore, e.g., a HTTP 301?" but that won't be reported as error anyway. So yes, `ignoreerrors` sounds good.
 
> For your use case though isn't the existing `ignore` enough? If these URLs are always failing with linkchecker?

Kind of but it'd be different. The linked content could go away completely and report 404 etc. In that case, the otherwise ignored error should turn back into a non-ignored error.

Speaking of such cases, there is one alternative that we might want to consider. Could we combine the solution of this issue with use cases where users expect certain results for certain URLs? For example ``http://example.com/private/*`` is *expected* to return ``401 Unauthorized``. This is different from the approach above, since this would also raise an error if the result is a non-error status. If you think this is out of scope (since linkchecker is no monitor in that sense), fine. If you think this would be a feature worthwhile implementing, would it be much harder to be implemented?

> Ultimately though, although it is fine to raise feature requests, new features are only likely to come about from PRs!

I am aware – the age of the ticket supports your statement well. :)

**Comment by cjmayo** [comment]
I think all the options are likely to be a reasonable piece of work as they are not just adapting the existing `ignore/ignorewarnings` but the only way to really find out is to try.

**Comment by lpirl** [comment]
Great, I'll try to look into this (i.e., linkchecker code base), hopefully next week.

**Comment by lpirl** [comment]
I made a drafty proposal (#629) and would be happy to receive some feedback.

**Comment by lpirl** [comment]
Since #629 has landed, should we close this ticket? Not sure if @keunes is still interested and would answer at some point.
